### PR TITLE
sql: support multi-statement reverse commands

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: sql/migrate/migrate.go [hunk 1]
```diff
@@ -46,15 +46,29 @@ type (
 		// A Comment describes the change.
 		Comment string
 
-		// Reverse contains the "reversed statement" if
-		// command is reversible.
-		Reverse string
+		// Reverse contains the "reversed" statement(s) if
+		// the command is reversible.
+		Reverse any // string | []string
```
Comment (masseelch) [inline sql/migrate/migrate.go:51]: Why not `[]string`? What code would we break by using `[]string`?

Comment (a8m) [reply sql/migrate/migrate.go:51]: All codebase uses `string` except for one case. I preferred `any` for this reason, but also there's a chance we'll add support for `*Change` in the future (e.g., adding a comment per statement).

File: sql/mysql/migrate.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func (p *planApply) PlanChanges(_ context.Context, name string, changes []schema
 		return nil, err
 	}
 	for _, c := range s.Changes {
-		if c.Reverse == "" {
+		if stmts, _ := c.ReverseStmts(); len(stmts) == 0 {
```
Comment (masseelch) [inline sql/mysql/migrate.go:43]: Why ignore the error? It is not guaranteed we have `string | []string` here, is it?

Comment (a8m) [reply sql/mysql/migrate.go:43]: I feel comfortable ignoring the error in this case because we build the changes in this file.

Comment (masseelch) [reply sql/mysql/migrate.go:43]: By using a `DiffHook` a user can add a `Reverse` that is not a `string` or `[]string`.

**Review by rotemtam — APPROVED** [review]
GJ

**Pull request merged**